test/components/views/settings/DevicesPanel-test.tsx

```diff
@@ -0,0 +1,195 @@
+
+            mockClient.getDevices.mockClear();
+
+            act(() => {
+                fireEvent.click(getByTestId('sign-out-devices-btn'));
+            });
+
+            expect(container.getElementsByClassName('mx_Spinner').length).toBeTruthy();
+            expect(mockClient.deleteMultipleDevices).toHaveBeenCalledWith([device2.device_id], undefined);
+
+            await flushPromises();
+
+            // devices refreshed
+            expect(mockClient.getDevices).toHaveBeenCalled();
+            // and rerendered
+            expect(container.getElementsByClassName('mx_DevicesPanel_device').length).toEqual(2);
+        });
+
+        it('deletes selected devices when interactive auth is required', async () => {
+            mockClient.deleteMultipleDevices
+                // require auth
+                .mockRejectedValueOnce(interactiveAuthError)
+                // then succeed
+                .mockResolvedValueOnce({});
+
+            mockClient.getDevices
+                .mockResolvedValueOnce({ devices: [device1, device2, device3] })
+                // pretend it was really deleted on refresh
+                .mockResolvedValueOnce({ devices: [device1, device3] });
+
+            const { container, getByTestId, getByLabelText } = render(getComponent());
+
+            await flushPromises();
+
+            // reset mock count after initial load
+            mockClient.getDevices.mockClear();
+
+            toggleDeviceSelection(container, device2.device_id);
+
+            act(() => {
+                fireEvent.click(getByTestId('sign-out-devices-btn'));
+            });
+
+            await flushPromises();
+            // modal rendering has some weird sleeps
+            await sleep(10);
+
+            const modal = document.getElementsByClassName('mx_Dialog');
+            expect(modal).toMatchSnapshot();
+
+            // fill password and submit for interactive auth
+            act(() => {
+                fireEvent.change(getByLabelText('Password'), { target: { value: 'topsecret' } });
+                fireEvent.submit(getByLabelText('Password'));
+            });
+
+            await flushPromises();
+
+            // devices refreshed
+            expect(mockClient.getDevices).toHaveBeenCalled();
+            // and rerendered
+            expect(container.getElementsByClassName('mx_DevicesPanel_device').length).toEqual(2);
+        });
+
+        it('clears loading state when interactive auth fail is cancelled', async () => {
+            mockClient.deleteMultipleDevices
+                // require auth
+                .mockRejectedValueOnce(interactiveAuthError)
+                // then succeed
+                .mockResolvedValueOnce({});
+
+            mockClient.getDevices
+                .mockResolvedValueOnce({ devices: [device1, device2, device3] })
+                // pretend it was really deleted on refresh
+                .mockResolvedValueOnce({ devices: [device1, device3] });
```

Comment:
Does this ever get called before we clear the mock,

Reply:
clearing the mock clears the mock counter, not the mock implementations (that's `mockReset`)

Reply:
Ah, sorry!